### ISSUE TITLE
Unable to publish after failed publish

### ISSUE THREAD

**Issue by mheffner** [npm/npm#7157]
I'm trying to publish an update of https://github.com/librato/statsd-librato-backend, version 0.1.5.

The first attempt to publish hung on the last step and was killed:
```
npm http PUT https://registry.npmjs.org/statsd-librato-backend/0.1.5/-tag/latest
^C
```

Attempts to run publish again fail with:
```
npm http PUT https://registry.npmjs.org/statsd-librato-backend/-/statsd-librato-backend-0.1.5.tgz/-rev/34-3dc090d212b07dd9992abd642bbbd08e
npm ERR! cb() never called!
npm ERR! not ok code 0
```

I updated to the latest npm and node v0.10.35, yet still unable to publish...this time the full backlog is:
```
npm ERR! publish Failed PUT 403
npm ERR! Error: forbidden Cannot replace existing tarball attachment: statsd-librato-backend
npm ERR!     at RegClient.<anonymous> (/home/mheffner/.nvm/v0.10.35/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:308:14)
npm ERR!     at Request._callback (/home/mheffner/.nvm/v0.10.35/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:246:65)
npm ERR!     at Request.self.callback (/home/mheffner/.nvm/v0.10.35/lib/node_modules/npm/node_modules/request/request.js:236:22)
npm ERR!     at Request.emit (events.js:98:17)
npm ERR!     at Request.<anonymous> (/home/mheffner/.nvm/v0.10.35/lib/node_modules/npm/node_modules/request/request.js:1142:14)
npm ERR!     at Request.emit (events.js:117:20)
npm ERR!     at IncomingMessage.<anonymous> (/home/mheffner/.nvm/v0.10.35/lib/node_modules/npm/node_modules/request/request.js:1096:12)
npm ERR!     at IncomingMessage.emit (events.js:117:20)
npm ERR!     at _stream_readable.js:944:16
npm ERR!     at process._tickCallback (node.js:442:13)
npm ERR! If you need help, you may report this *entire* log,
npm ERR! including the npm and node versions, at:
npm ERR!     <http://github.com/npm/npm/issues>

npm ERR! System Linux 3.17.4-200.fc20.x86_64
npm ERR! command "/home/mheffner/.nvm/v0.10.35/bin/node" "/home/mheffner/.nvm/v0.10.35/bin/npm" "publish"
npm ERR! cwd /home/mheffner/git/librato/statsd-librato-backend
npm ERR! node -v v0.10.35
npm ERR! npm -v 1.4.28
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/mheffner/git/librato/statsd-librato-backend/npm-debug.log
npm ERR! not ok code 0
```

It's unclear what the fix for this is. I imagine it has something to do with the failed partial ship.

**Comment by othiym23** [comment]
I think that version is pretty much dead. It looks to me like the original publish attempt was done with a version of npm less than 1.4, which predates the single-step publish process the npm registry has supported (and pretty much required) for the last year. So the publish is permanently stuck in an in-between state, unless you feel like writing to support@npmjs.com and asking them / us to delete the errant tarball.

The easiest thing to do at this point is to

1. update to the actual latest version of npm, which is `npm@2.2.0`. The version shipped with Node has been stuck at `1.4.28` since the release of `npm@2.0.0` (the why of that is a question for Joyent -- it is not the npm team's choice).
2. `npm version patch && npm publish`. The next version of `statsd-librato-backend` is now 0.1.6!

If you take away anything from this experience, let it be that trying to publish packages with old versions of npm is dangerous, and if you must use an older version of npm, make sure it's at least 1.4.0. Sorry this has turned out to be such a pain.

**Comment by mheffner** [comment]
@othiym23 Thanks for the detailed response! I've gone ahead and simply bumped it to 0.1.6 and will cut releases with 2.2.0 in the future.
